### PR TITLE
1560 - Apply area and address specific rental property rules when creating applicant

### PR DESCRIPTION
**Pull request by momentiris**
- Adds property and residential validation when creating applicant for listing

**TODO:** 
- Add general error handling on the contact query endpoint consuming component
- Handle all responses from core. 403, 404 etc.
~- Generalise Tenant type. Its currently copied from leasing~

### CODE REVIEW

**Review by driatic — APPROVED** [review]
Super! Go a head and merge when changes in leasing and core are implemented.

File: packages/backend/src/services/leasing-service/adapters/core-adapter.ts [hunk 1]
```diff
@@ -64,13 +65,23 @@ const getContactsDataBySearchQuery = async (
   }
 }
 
-const getContactByContactCode = async (
+type NonEmptyArray<T> = [T, ...T[]]
+
+export type Tenant = Omit<Contact, 'leases' | 'isTenant'> & {
```
[the remark this answers is not in the document]
Comment (momentiris) [reply packages/backend/src/services/leasing-service/adapters/core-adapter.ts]: https://github.com/Bostads-AB-Mimer/onecore-internal-portal/pull/30/commits/ee46276deafa9bec8204f386b5c77019b73a9635

[the remark this answers is not in the document]
Comment (momentiris) [reply packages/backend/src/services/leasing-service/adapters/core-adapter.ts]: https://github.com/Bostads-AB-Mimer/onecore-internal-portal/pull/30/commits/97b63d4fc521260e262a8cbf9104e08e29832913

**Review by driatic — APPROVED** [review]
🚀

**Pull request merged**